src/test/java/com/synopsys/integration/detect/battery/docker/GoModTest.java

```diff
@@ -0,0 +1,90 @@
+package com.synopsys.integration.detect.battery.docker;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import com.synopsys.integration.detect.battery.docker.integration.BlackDuckAssertions;
+import com.synopsys.integration.detect.battery.docker.integration.BlackDuckTestConnection;
+import com.synopsys.integration.detect.battery.docker.provider.BuildDockerImageProvider;
+import com.synopsys.integration.detect.battery.docker.util.DetectCommandBuilder;
+import com.synopsys.integration.detect.battery.docker.util.DetectDockerTestRunner;
+import com.synopsys.integration.detect.battery.docker.util.DockerAssertions;
+import com.synopsys.integration.detect.configuration.DetectProperties;
+import com.synopsys.integration.detect.configuration.enumeration.DetectTool;
+import com.synopsys.integration.detector.base.DetectorType;
+import com.synopsys.integration.exception.IntegrationException;
+
+@Tag("integration")
+public class GoModTest {
+
+    private static final String[] GO_VERSIONS_TO_TEST = new String[] {
+        "1.16.15",
+        "1.17.13",
+        "1.18.10",
+        "1.19.12",
+        "1.20.4"
```

Comment:
Included latest patch version for each release except for Go 1.20 which is up to 1.20.4 (latest Go version is 1.21 and within Go 1.20, latest patch version is 1.20.7)